### PR TITLE
KafkaInputOperator fixes.

### PR DESCRIPTION
**Pull request by tweise**

### CODE REVIEW

**Comment by tweise** [comment]
I also noticed that the tests don't pass:

contrib$ mvn test -Dtest=KafkaInputOperatorTest -DskipTests=false

Tests run: 4, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 73.123 sec <<< FAILURE!
testRecoveryAndIdempotency(com.datatorrent.contrib.kafka.KafkaInputOperatorTest)  Time elapsed: 12.975 sec  <<< FAILURE!
java.lang.AssertionError: largest recovery window expected:<2> but was:<3>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:743)
	at org.junit.Assert.assertEquals(Assert.java:118)
	at org.junit.Assert.assertEquals(Assert.java:555)
	at com.datatorrent.contrib.kafka.KafkaInputOperatorTest.testRecoveryAndIdempotency(KafkaInputOperatorTest.java:257)


Results :

Failed tests: 
  KafkaInputOperatorTest.testRecoveryAndIdempotency:257 largest recovery window expected:<2> but was:<3>

File: contrib/src/main/java/com/datatorrent/contrib/kafka/AbstractKafkaInputOperator.java [hunk 1]
```diff
@@ -392,13 +367,11 @@ public void emitTuples()
...
-      {
-        currentWindowRecoveryState.put(message.kafkaPart, new KafkaPair<Long, Integer>(message.offSet, 1));
+      MutablePair<Long, Integer> offsetAndCount = currentWindowRecoveryState.get(message.kafkaPart);
+      if(offsetAndCount == null) {
+        currentWindowRecoveryState.put(message.kafkaPart, new MutablePair<Long, Integer>(message.offSet, 1));
       } else {
-        Pair<Long, Integer> second = currentWindowRecoveryState.get(message.kafkaPart);
-        Integer noOfMessages = second.getSecond();
-        currentWindowRecoveryState.put(message.kafkaPart, new KafkaPair<Long, Integer>(second.getFirst(), noOfMessages+1));
+        offsetAndCount.setRight(offsetAndCount.right++);
```
Comment (chaithu14) [inline contrib/src/main/java/com/datatorrent/contrib/kafka/AbstractKafkaInputOperator.java]: Post/Pre-increment doesn't work for Integer data type.

**Comment by chaithu14** [comment]
Please remove the oldest recovery path(i.e, target) directory under Malhar and then run the test case.

**Comment by tweise** [comment]
This test still does not pass (after changing increment and clean):

Results :

Failed tests: 
  KafkaInputOperatorTest.testRecoveryAndIdempotency:271 Total messages collected  expected:1500 but was:1000

Tests run: 4, Failures: 1, Errors: 0, Skipped: 0

Further the test should not require a manual clean step. It should clean any leftover state in setup.

**Pull request merged**